include/cute/arch/util.hpp

```diff
@@ -114,7 +114,7 @@ cast_smem_ptr_to_uint(void const* const ptr)
 
 #elif defined(CUTLASS_ENABLE_SYCL)
 
-    return (intptr_t)(__attribute__((opencl_local)) void const *)ptr;
+  return (intptr_t)(sycl::decorated_local_ptr<const void>::pointer)ptr;
```

Comment:
```suggestion
  return (uint32_t)(sycl::decorated_local_ptr<const void>::pointer)ptr;
```
Shared memory pointers are always expected to be within 32 bits. `intptr_t` is defined as " integer type capable of holding a pointer " , which is 64 bit I suppose. 

All instructions taking the smem pointer using the "r" type, which denotes a 32 bit data ([doc](https://docs.nvidia.com/cuda/inline-ptx-assembly/index.html#constraints)), for example

```asm
#if defined(CUTE_ARCH_TMA_SM90_ENABLED)
    uint32_t smem_int_ptr  = cast_smem_ptr_to_uint(smem_ptr);
    asm volatile("cp.async.bulk.global.shared::cta.bulk_group [%0], [%1], %2;\n"
                     :
                     : "l"(gmem_ptr), "r"(smem_int_ptr), "r"(store_bytes)
                     : "memory");
```

Do you think using intptr_t can potentially lead to issues ?

Reply:
Nevermind, 
every where this is used, the smem_int_ptr is marked as uint32_t,
so the conversion happens either way

Resolving

Reply:
Should be the same but `uint32_t` is feel more correct. Good catch!

Reply:
The compiler does not like the direct conversion...

Reply:
oh yes, 

that would be because static_cast from pointer to integer would not be allowed 🤦🏻 .
Sorry about that,

Well, do we really need especial SYCL path ?
just below that, there is 
```asm
#elif defined(__CUDA_ARCH__)

  uint32_t smem_ptr;

  asm(
  "{ .reg .u64 smem_ptr; cvta.to.shared.u64 smem_ptr, %1; cvt.u32.u64 %0, smem_ptr; }\n" 
    : "=r"(smem_ptr) : "l"(ptr));

  return smem_ptr;
```

which is what happens under the hood,
we can change that to 
```asm
#elif defined(__CUDA_ARCH__) || (__SYCL_CUDA_ARCH__) // even this will go away after cuda compatibility flag
```

and do away with the SYCL way of doing it